src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala

```diff
@@ -278,10 +286,94 @@ class NeedlemanWunschAlignerTest extends UnitSpec {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Local).align(s("ACGTAACC"), s("ACGTAACC"))
+    assertValidLocalAlignment(result)
+    result.cigar.toString() shouldBe "8="
+    result.score shouldBe 8
+  }
+
+  it should "correctly align an identical subsequence (query in target)" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Local).align(s("CCGG"), s("AACCGGTT"))
+    assertValidLocalAlignment(result)
+    result.queryStart shouldBe 1
+    result.targetStart shouldBe 3
+    result.queryEnd shouldBe 4
+    result.targetEnd shouldBe 6
+    result.cigar.toString() shouldBe "4="
+    result.score shouldBe 4
+  }
+
+  it should "correctly align an identical subsequence (target in query)" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Local).align(s("AACCGGTT"), s("CCGG"))
+    assertValidLocalAlignment(result)
+    result.queryStart shouldBe 3
+    result.targetStart shouldBe 1
+    result.cigar.toString() shouldBe "4="
+    result.score shouldBe 4
+  }
+
+  it should "correctly align a sub-sequence with a leading mismatch" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Local).align(s("AGCGTCGTATACGTCGTA-------"), s("CGCGTCGTATACGTCGTAAAGATAT"))
+    assertValidLocalAlignment(result)
+    result.queryStart shouldBe 2
+    result.targetStart shouldBe 2
+    result.queryEnd shouldBe 18
+    result.targetEnd shouldBe 18
+    result.cigar.toString() shouldBe "17="
+    result.score shouldBe 17
+  }
+
+
+  it should "correctly align a sub-sequence with a trailing mismatch" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Local).align(s("-------CGCGTCGTATACGTCGTT"), s("AAGATATCGCGTCGTATACGTCGTA"))
+    assertValidLocalAlignment(result)
+    result.queryStart shouldBe 1
+    result.targetStart shouldBe 8
+    result.queryEnd shouldBe 17
+    result.targetEnd shouldBe 24
+    result.cigar.toString() shouldBe "17="
+    result.score shouldBe 17
+  }
+
+  it should "correctly align a sub-sequence with a gap in the query" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Local).align(s("CCGCGCGCGC"), s("AACCGCGACGCGCTT"))
+    assertValidLocalAlignment(result)
+    result.queryStart shouldBe 1
+    result.targetStart shouldBe 3
+    result.queryEnd shouldBe 10
+    result.targetEnd shouldBe 13
+    result.cigar.toString() shouldBe "5=1D5="
+    result.score shouldBe 6
+  }
+
+  it should "correctly align a sub-sequence with a gap in the target" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Local).align(s("AACCGCGACGCGCTT"), s("CCGCGCGCGC"))
+    assertValidLocalAlignment(result)
+    result.queryStart shouldBe 3
+    result.targetStart shouldBe 1
+    result.queryEnd shouldBe 13
+    result.targetEnd shouldBe 10
+    result.cigar.toString() shouldBe "5=1I5="
+    result.score shouldBe 6
+  }
+
+  it should "correctly align prefer a match over an indel" in {
```

Comment:
this is implementation specific, where we check the max score is from the diagonal before up/down.

Reply:
Sounds good.  The name of the test is a little off though, perhaps just remove "correctly align".